src/store/modules/ProfilesModule.ts

```diff
@@ -0,0 +1,51 @@
+import FileUtils from "../../utils/FileUtils";
+import R2Error from "../../model/errors/R2Error";
+import { ActionTree } from "vuex";
+import { State as RootState } from "../../store";
+import Profile from "../../model/Profile";
+import FsProvider from "../../providers/generic/file/FsProvider";
+
+interface State {
+    profileList: string[];
+}
+
+/**
+ * State for Profiles, i.e. list for profiles in a single game/community.
+ */
+export const ProfilesModule = {
+    namespaced: true,
+
+    state: (): State => ({
+        profileList: ['Default'],
+    }),
+    mutations: {
+        setProfileList(state: State, profileList: string[]) {
+            state.profileList = profileList;
+        },
+    },
+    actions: <ActionTree<State, RootState>>{
+        async removeSelectedProfile({rootGetters, state, dispatch}) {
+            const activeProfile: Profile = rootGetters['profile/activeProfile'];
+            const path = activeProfile.getPathOfProfile();
+            const profileName = activeProfile.getProfileName();
+
+            try {
+                await FileUtils.emptyDirectory(path);
+                await FsProvider.instance.rmdir(path);
+            } catch (e) {
+                throw R2Error.fromThrownValue(e, 'Error whilst deleting profile from disk');
+            }
+
+            if (profileName.toLowerCase() !== 'default') {
+                state.profileList = state.profileList.filter((p: string) => p !== profileName)
+                await dispatch('setSelectedProfile', 'Default');
+            }
+        },
+
+        async setSelectedProfile({rootGetters, state, dispatch}, profileName: string) {
```

Comment:
This is fine now, but for RenameProfileModal this is going to need to support the additional `prewarmCache` parameter which determines whether the last two dispatches are executed (cf. `Profiles.setSelectedProfile`).

This is because the operation is quite heavy for profiles with hundreds of mods, and we don't want to do it unnecessarily. You can easily test it by installing a few mod packs for Lethal Company and switching between the profiles in the list.

Anyway, it's not necessary for this PR so it's up to you whether you want to do it here or in the RenameProfileModal PR.